events/api/build.gradle.kts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import io.smallrye.openapi.api.OpenApiConfig.OperationIdStrategy
+import io.smallrye.openapi.gradleplugin.SmallryeOpenApiExtension
+import io.smallrye.openapi.gradleplugin.SmallryeOpenApiTask
+
+plugins {
+  `java-library`
+  jacoco
+  `maven-publish`
+  signing
+  `nessie-conventions`
+  alias(libs.plugins.smallrye.openapi)
+}
+
+extra["maven.name"] = "Nessie - Events - API"
+
+dependencies {
+  implementation(project(":nessie-model"))
+
+  // Immutables
+  implementation(libs.immutables.builder)
+  implementation(libs.immutables.value.annotations)
+  annotationProcessor(libs.immutables.value.processor)
+
+  // Open API
+  implementation(libs.microprofile.openapi)
+
+  // Jackson
+  implementation(platform(libs.jackson.bom))
+  implementation(libs.jackson.annotations)
+  implementation(libs.jackson.databind)
+
+  // ErrorProne
+  implementation(libs.errorprone.annotations)
+
+  // Testing
+  testImplementation(platform(libs.junit.bom))
+  testImplementation(libs.bundles.junit.testing)
+  testImplementation(libs.junit.jupiter.engine)
+  testImplementation(libs.jackson.datatype.jdk8)
+  testImplementation(libs.jackson.datatype.jsr310)
+}
+
+extensions.configure<SmallryeOpenApiExtension> {
+  scanDependenciesDisable.set(false)
+  infoVersion.set(project.version.toString())
+  schemaFilename.set("META-INF/openapi/openapi")
+  operationIdStrategy.set(OperationIdStrategy.METHOD)
+  scanPackages.set(listOf("org.projectnessie.events.spi"))
+  scanClasses.set(
+    listOf(
+      "org.projectnessie.model.Content",
```

Comment:
These are the model classes referenced by the events API. Basically, `Content` and its subtypes.

I wonder though:

1. Should we make this module completely independent from `nessie-model`?
2. Will this work in the presence of custom content types? I think so, but we can re-evaluate this after #6618. The only issue for custom content types would be for users: to deserialize a custom serialized content, they would need the concrete implementation.

Reply:
They wouldn't need the concrete implementation to deserialize an unknown content-type - they'd get a `GenericContent` object as a fallback (that particular `GenericContent` class shouldn't be used directly though)

Reply:
I need to test, but my fear is that a Kafka consumer may fail to deserialize a `GenericContent` because `GenericContent` is not part of the schema.

More specifically, [here](https://github.com/confluentinc/schema-registry/blob/v7.5.0-532/json-schema-serializer/src/main/java/io/confluent/kafka/serializers/json/AbstractKafkaJsonSchemaDeserializer.java#L178) the Kafka deserializer attempts to validate the schema. Even worse: [here](https://github.com/confluentinc/schema-registry/blob/v7.5.0-532/json-schema-serializer/src/main/java/io/confluent/kafka/serializers/json/AbstractKafkaJsonSchemaDeserializer.java#L249), it will attempt to locate the exact subtype among the `oneOf` alternatives available. These operations will fail if the subtype is not explicitly listed.

(This Kafka deserializer doesn't honor annotations such as `@JsonTypeIdResolver`.)

I think it would be safer to cut the dependency to nessie:model completely and provide a simplified `Content` interface here. Wdyt?

Reply:
> cut the dependency to nessie:model completely

SGTM